worker/src/RTC/RtpPacket.cpp

```diff
@@ -600,12 +598,94 @@ namespace RTC
 
 			auto currentLen = extension->len + 1;
 
-			// Fill with 0's if new length is minor.
-			if (len < currentLen)
-				std::memset(extension->value + len, 0, currentLen - len);
+			MS_DEBUG_DEV(
+			  "set extension id: %" PRIu8 ", length:%" PRIu8 ", currentLen:%" PRIu8 " value:%s",
+			  id,
+			  len,
+			  currentLen,
+			  value.c_str());
+			
+			if (len != currentLen)
+			{
+				// If len != currentLen, means we may need to shift extension or payload.
+				uint8_t* extensionStart    = reinterpret_cast<uint8_t*>(this->headerExtension) + 4;
+				uint8_t* extensionEnd      = extensionStart + GetHeaderExtensionLength();
+				uint8_t* ptr               = extensionStart;
+				size_t extensionsTotalSize = static_cast<size_t>(len + 1);
+				uint8_t* ptr1              = ptr;
+				// Clear current extension, we will rewrite after move all extension in right place.
+				std::memset((uint8_t*)extension, 0, currentLen + 1);
+				while (ptr < extensionEnd || ptr1 < extensionEnd)
+				{
+					if (ptr >= extensionEnd)
+					{
+						*ptr1 = 0u;
+						ptr1++;
+						continue;
+					}
+					const uint8_t tempId = (*ptr & 0xF0) >> 4;
+					const size_t tempLen = static_cast<size_t>(*ptr & 0x0F) + 1;
 
-			// In One-Byte extensions value length 0 means 1.
-			extension->len = len - 1;
+					// TempId=15 in One-Byte extensions means "stop parsing here".
+					if (tempId == 15u)
+						break;
+					if (tempId != 0u && tempId != id)
+					{
+						if (ptr1 < ptr)
+						{
+							std::memmove(ptr1, ptr, tempLen + 1);
+							this->oneByteExtensions[tempId - 1] = reinterpret_cast<OneByteExtension*>(ptr1);
+						}
+						extensionsTotalSize += tempLen + 1;
+						ptr += tempLen + 1;
+						ptr1 += tempLen + 1;
+				
+						MS_DEBUG_DEV(
+						  "tempId: %" PRIu8 " tempLen:%zd, offset:%ld", tempId, tempLen, ptr - extensionStart);
+					
+					}
+					else
```

Comment:
This `else` in `ParseExtensions()` is only reachable for `tempId == 0.` Now it's reachable for `tempId == 0` and `tempId == id`